src/handlebars-utils.js

```diff
@@ -128,7 +128,8 @@ HandlebarsUtils.isValidExpression = function(input, i, type) {
                     re.result = false;
                     return re;
                 }
-                if (re[2] === '') {
+                // re[2] being empty and [a b] in {{[a b]}} are both regarded as a single ID
+                if (re[2] === '' || (re[1].indexOf('[') === 0 && /\]\s*$/.test(re[2]))) {
```

Comment:
as the lex definition is <mu>'['[^\]]*']', the code cannot cover more complicated case like [a b c d ... ]